umm-spec-lib/src/cmr/umm_spec/validation/platform.clj

```diff
@@ -34,24 +34,43 @@
    with the model number. So the short name is not specific enough to identify
    specific platform instances. Therefore we are adding Characteristic names and values
    so that specific aircraft tail numbers, or boat IDS, etc. can be specified with the
-   short name."
+   short name. The |-| string is also added as a delimiter between the fields. This is used
+   in the getErrorString parameter-parser to parse out the different fields, so that a
+   meaningful error string can be suppplied back to the user."
   [values unique-fields]
   (let [{:keys [ShortName Characteristics Name Value]} unique-fields]
     (map (fn [value]
            (apply str
                  (ShortName value)
-                 (map #(str (Name %) (Value %)) (Characteristics value))))
+                 "|-|"
+                 (map #(str (Name %) "|-|" (Value %)) (Characteristics value))))
          values)))
 
+(defn- getErrorString
+  "Creates the error string for platforms depending on which elements (ShortName, Characteristic/Name, Characteristic/Value) are present.
+   The element order is ShortName, Name, Value. The values are separated by the |-| string."
+  [duplicate-name]
+  (let [platformElements (clojure.string/split duplicate-name #"\|-\|")
+        platCounts (count platformElements)]
```

Comment:
Nit: Rather than calling `(get platformElements {index})` a bunch, it might be worth adding this to the `let` to clean things up a bit
```
short-name (first platform-elements)
characteristic-name (second platform-elements)
characteristic-value (last platform-elements)
```